src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Attr.java

```diff
@@ -1419,9 +1419,10 @@ public void visitBlock(JCBlock tree) {
             // Block is a static or instance initializer;
             // let the owner of the environment be a freshly
             // created BLOCK-method.
+            Type fakeOwnerType = new MethodType(List.nil(), syms.voidType, List.nil(), syms.methodClass);
             Symbol fakeOwner =
                 new MethodSymbol(tree.flags | BLOCK |
-                    env.info.scope.owner.flags() & STRICTFP, names.empty, null,
+                    env.info.scope.owner.flags() & STRICTFP, names.empty, fakeOwnerType,
```

Comment:
the type was left null here, which caused some crashes in `LambdaToMethod`. There was really no reason for this to be null, except that `ClassWriter` looks at type being null as a test to see if owner is a static/instance init block. But we can use `BLOCK` for that (as `TypeAnnotations` does)